index.js

```diff
@@ -1,14 +1,90 @@
 require('dotenv').config();
 const express = require('express');
 const app = express();
+const cors = require('cors');
 const axios = require('axios');
-const ClientOAuth2 = require('client-oauth2')
+const cookie_parser = require('cookie-parser')
 
 app.set('view engine', 'ejs');
 app.use(express.urlencoded({ extended: true }));
 app.use(express.json());
+app.use(cors('*'))
+app.use(cookie_parser());
 app.use(express.static('public'));
 
+app.set('trust proxy', 1);
+
+const authConfig = {
+  providers: [
+    {
+      id: 'signalwire',
+      name: 'SignalWire',
+      type: 'oauth',
+      authorization: {
+        url: process.env.OAUTH_AUTH_URI,
+        params: { scope: 'email' }
+      },
+      clientId: process.env.OAUTH_CLIENT_ID,
+      clientSecret: process.env.OAUTH_SECRET,
+      token: process.env.OAUTH_TOKEN_URI,
+      userinfo: process.env.OAUTH_USERINFO_URI,
+      profile(profile) {
+        return {
+          id: profile.id,
+          email: profile.email,
+          first_name: profile.first_name,
+          last_name: profile.last_name,
+          display_name: profile.display_name,
+          job_title: profile.job_title,
+          push_notification_key: profile.push_notification_key
+        };
+      }
+    }
+  ],
+  callbacks: {
+    jwt: ({ token, account, profile }) => {
+      if (account) {
+        token.accessToken = account.access_token
+        token.id = profile.id
+        token.pushNotificationKey = profile.push_notification_key
+      }
+      return token
+    },
+    session({ session, token }) {
+      return {
+        ...session,
+        sat: token.accessToken,
+        pushNotificationKey: token.pushNotificationKey,
+        user: {
+          id: token.id,
+          ...session.user,
+          username: session.user.email
+        }
+      }
+    }
+  }
+}
+
+let authGetSession
+
+import('@auth/express').then(({ ExpressAuth, getSession }) => {
+  authGetSession = getSession
+  async function authSession(req, res, next) {
+    res.locals.session = await getSession(req, authConfig)
+    next()
+  }
+
+
+
+  app.use('/api/auth/*', ExpressAuth(authConfig))
+  app.use(authSession)
+
+  app.listen(process.env.PORT || 3000, () => {
```

Comment:
currently and unfortunately,  yes...

I agreed with that most existing code at this point don't feature that. But that is because this esm is getting more popular now. And there is a strong push for being the default.

But I guess we could also make our code an es module as better alternative